src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java

```diff
@@ -259,28 +288,96 @@ private VersionNumberBuildInfo incBuild(AbstractBuild build, PrintStream log) th
             // increment total builds
             buildsAllTime = info.getBuildsAllTime() + buildInc;
         }
+        
         // have we overridden any of the version number info?  If so, set it up here
         boolean saveOverrides = false;
-        if (this.oBuildsToday >= 0) {
-            buildsToday = oBuildsToday;
-            oBuildsToday = -1;
-            saveOverrides = true;
+        Pattern pattern = Pattern.compile(envVarPattern);
```

Comment:
Yes, I agree. It would be great to rename <code> envVarPattern</code> to uppercase, but it's up to you (no explicit coding style definition for the plugin)